### ISSUE TITLE
Sudden mode collapse after 400k epochs

### ISSUE THREAD

**Issue by stefdoerr** [carpedm20/BEGAN-tensorflow#19]
I have had this weird behaviour where after 380000 epochs the generator suddenly mode collapses.
For example see:

epoch 350k
![349500_g](https://cloud.githubusercontent.com/assets/7935362/25562745/92346624-2d8e-11e7-8ba6-09d4b11823da.png)

epoch 390k
![389500_g](https://cloud.githubusercontent.com/assets/7935362/25562750/a5936814-2d8e-11e7-8acb-873f25b9e6e8.png)

epoch 396k
![396000_g](https://cloud.githubusercontent.com/assets/7935362/25562755/afe2587a-2d8e-11e7-9258-48603bdb465c.png)

epoch 400k
![402000_g](https://cloud.githubusercontent.com/assets/7935362/25562758/b87ccb3c-2d8e-11e7-87f7-cd1bbf9eba2a.png)

epoch 480k
![483000_g](https://cloud.githubusercontent.com/assets/7935362/25562765/d02af218-2d8e-11e7-97a0-3a05792f7c64.png)

I am suspecting that portrait guy in the middle of the lower row as the poltergeist that caused the mode collapse but I can't prove it :laughing: . Anyone seen this happen before? 

You can also see the g_loss and d_loss_fake increasing around epoch 380k
![screenshot from 2017-04-30 10-23-16](https://cloud.githubusercontent.com/assets/7935362/25562791/3e93dada-2d8f-11e7-94c4-f3609b970783.png)

**Comment by carpedm20** [comment]
I guess reducing learning rate too much (too small lr) hurts the training and also occurs mode collapsing. I'd better set a lowest boundary of learning rate to solve this problem.

**Comment by carpedm20** [comment]
Please try the most recent code and let me know if the same problem occurs.

**Comment by stefdoerr** [comment]
Sorry I think the error was on my side since I modified some of the code and probably introduced a mistake. Your fix didn't help which confirms it's probably on my side. You might want to undo the change if you think it affects the results negatively.

**Comment by stefdoerr** [comment]
Ok I think I found the issue. I will have to retrain to make sure but I'm quite convinced.

In this line
https://github.com/carpedm20/BEGAN-tensorflow/blob/master/models.py#L19
you were passing all variables, not just the trainable ones. I had modified a bit the code for the nearest-neighbour upscaling and added a new variable which I didn't want to train.
I replaced it now with `get_trainable_variables`

**Comment by stefdoerr** [comment]
Indeed the error was on my side. By using only the trainable variables now my weights stayed fixed and I got the same results. You can undo the thresholding commit if you want.

**Comment by dougsouza** [comment]
@stefdoerr, I'm having the same issue. I don't understand, `why tf.contrib.framework.get_variables(scope)` would be a problem since all the variables in this scope are treinable? What weights shouldn't be update?

**Comment by stefdoerr** [comment]
@dougsouza I had modified the code and added some non-trainable variables which I then accidentally trained. I guess the original code doesn't have any.

**Comment by dougsouza** [comment]
@stefdoerr, oh I understand. I'm running the original code and experiencing mode collapse after 400k steps as well, for the model that generates 64x64 images. For the model that generates 128x128 images it is happening after 300k steps.
